src/App.tsx

```diff
@@ -1,78 +1,40 @@
-/**
- * Sample React Native App
- * https://github.com/facebook/react-native
- *
- * Generated with the TypeScript template
- * https://github.com/react-native-community/react-native-template-typescript
- *
- * @format
- */
+import React, { useState, useEffect } from 'react';
+import { View, Text, StyleSheet } from 'react-native';
+import { observer } from 'mobx-react-lite';
 
-import React, {useState, useEffect} from 'react';
-import {
-  SafeAreaView,
-  StyleSheet,
-  ScrollView,
-  View,
-  Text,
-  StatusBar,
-} from 'react-native';
-
-import {
-  Header,
-  LearnMoreLinks,
-  Colors,
-  DebugInstructions,
-  ReloadInstructions,
-} from 'react-native/Libraries/NewAppScreen';
+import { firebaseService } from './services';
+import { useStores } from './hooks';
 
+import Chat from './components/Chat';
 import Button from './components/common/Button';
-import { AppConfig } from './AppConfig';
-
-import auth, { firebase } from '@react-native-firebase/auth';
-//import styles from './components/Input/styles';
-
-declare var global: {HermesInternal: null | {}};
 
-const App = () => {
+const App = observer(() => {
+  const { userStore } = useStores()
   const [initializing, setInitializing] = useState(true);
-  const [user, setUser] = useState();
-  const [firebaseApp, setFirebaseApp] = useState();
 
   function onAuthStateChanged(user: any): void {
-    setUser(user);
+    userStore.setCurrentUser(user);
     if (initializing) setInitializing(false);
   }
 
-  function maybeInitializeFirebase(): void {
-    try {
-      if (!firebase.app()) {
-        firebase.initializeApp(AppConfig);
-      }
-    } catch (e) {
-      firebase.initializeApp(AppConfig);
-    }
-  }
-
   useEffect(() => {
-    maybeInitializeFirebase();
-    const subscriber = auth().onAuthStateChanged(onAuthStateChanged);
+    const subscriber = firebaseService.auth.onAuthStateChanged(onAuthStateChanged);
     return subscriber; // unsubscribe on unmount
   }, []);
```

Comment:
Sorry, could you uninstall, then reinstall the app on your phone and make sure that Firebase is being initialised properly on first start? If I clear the entire cache on my emulator, or reinstall the app, I get the 'No Firebase App' error message on your branch.

![image](https://user-images.githubusercontent.com/7235102/74819580-78f37200-52b5-11ea-930c-5062b2ff65d1.png)